src/main/java/de/qabel/desktop/storage/command/CreateFolderChange.java

```diff
@@ -0,0 +1,29 @@
+package de.qabel.desktop.storage.command;
+
+import de.qabel.core.crypto.CryptoUtils;
+import de.qabel.desktop.exceptions.QblStorageException;
+import de.qabel.desktop.storage.BoxFolder;
+import de.qabel.desktop.storage.DirectoryMetadata;
+import org.spongycastle.crypto.params.KeyParameter;
+
+public class CreateFolderChange implements DirectoryMetadataChange<ChangeResult<BoxFolder>> {
+	private CryptoUtils cryptoUtils = new CryptoUtils();
+	private String name;
+	private byte[] deviceId;
+	private final KeyParameter secretKey;
+
+	public CreateFolderChange(String name, byte[] deviceId) {
+		this.name = name;
+		this.deviceId = deviceId;
+		secretKey = cryptoUtils.generateSymmetricKey();
```

Comment:
No need to save the secretKey since it is private and _only_ used in the execute method.